### PR TITLE
chore: Implement gRPC and HttpJson showcase tests for IAM

### PR DESCRIPTION
**Pull request by lqiu96**
Thank you for opening a Pull Request! For general contributing guidelines, please refer to [contributing guide](https://github.com/googleapis/gapic-generator-java/blob/main/CONTRIBUTING.md)

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/gapic-generator-java/issues/new/choose) before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Part of #1439 ☕️

These showcase tests were modeled after gapic-generator-go's IAM showcase tests.

### CODE REVIEW

File: showcase/gapic-showcase/pom.xml [hunk 1]
```diff
@@ -229,5 +229,10 @@
       <classifier>testlib</classifier>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>grpc-google-iam-v1</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (lqiu96) [inline showcase/gapic-showcase/pom.xml:236]: Added since there are compilation issues for MockIAMPolicy and MockIAMPolicyImpl

File: showcase/BUILD.bazel [hunk 2]
```diff
@@ -19,7 +19,8 @@ proto_library_with_info(
         "@com_google_gapic_showcase//schema/google/showcase/v1beta1:showcase_proto",
 #        "//showcase/gapic-showcase-extended/proto:showcase_proto_extended",
         "@com_google_googleapis//google/cloud:common_resources_proto",
-        "@com_google_googleapis//google/cloud/location:location_proto"
+        "@com_google_googleapis//google/cloud/location:location_proto",
+        "@com_google_googleapis//google/iam/v1:iam_policy_proto"
```
Comment (blakeli0) [inline showcase/BUILD.bazel:23]: So adding this `iam_policy_proto` implicitly enables IAM feature? Is this how we are suppose to enable IAM?

File: showcase/pom.xml [hunk 3]
```diff
@@ -80,6 +80,11 @@
               <!-- TODO(1547): Remove this exclusion once MessagingClientHttpJsonTest#updateBlurbTest is fixed -->
               <excludes>
                 <exclude>**/MessagingClientHttpJsonTest.java</exclude>
+                <exclude>**/ComplianceClientHttpJsonTest.java</exclude>
```
Comment (blakeli0) [inline showcase/pom.xml:83]: Are these changes related?

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+  @BeforeClass
+  public static void createClients() throws Exception {
+    grpcClient = TestClientInitializer.createGrpcIdentityClient();
+    httpjsonClient = TestClientInitializer.createHttpJsonIdentityClient();
+  }
+
+  @Before
+  public void setupTests() {
+    resourceName = "users/" + UUID.randomUUID().toString().substring(0, 8);
```
Comment (blakeli0) [inline showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:57]: Why the resource name is in this format? I see that it is a wild card resource based on the definition in [googleapis](https://github.com/googleapis/googleapis/blob/bd3fbf4776fe1240f2fddcc525a6b6e7690232ad/google/iam/v1/iam_policy.proto#L106), I guess it is randomly chosen?

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+            .build();
+    Policy policy = httpjsonClient.setIamPolicy(policyRequest);
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testGrpc_setIamPolicy_missingResource() {
+    SetIamPolicyRequest setIamPolicyRequest = SetIamPolicyRequest.newBuilder().build();
+    assertThrows(
+        InvalidArgumentException.class, () -> grpcClient.setIamPolicy(setIamPolicyRequest));
```
Comment (blakeli0) [inline showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:96]: Where is this exception being thrown? Is it in our client libraries? I see that we are adding a lot of error scenarios in this IT, a general comment is that we only want to test the exception if it's being thrown in the IAM code before it reaches backend, or if there is any specific response exception mapping logic in IAM code.

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+        httpjsonClient.getIamPolicy(
+            GetIamPolicyRequest.newBuilder().setResource(policyRequest.getResource()).build());
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testGrpc_getIamPolicy_doesNotExist() {
+    GetIamPolicyRequest getIamPolicyRequest =
+        GetIamPolicyRequest.newBuilder().setResource(resourceName).build();
+    assertThrows(NotFoundException.class, () -> grpcClient.getIamPolicy(getIamPolicyRequest));
```
Comment (blakeli0) [inline showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:156]: I don't think we want to test this scenario, as this exception is based on a backend response. For example, not likely but the backend could return a different status code which will be mapped to a different exception.

File: showcase/BUILD.bazel (hunk 2 above)
Comment (lqiu96) [reply showcase/BUILD.bazel:23]: I believe having this in the bazel rule enables it: https://github.com/googleapis/gapic-showcase/blob/b146f892d98ea0d733c45a2315941e8e6ab69c90/schema/google/showcase/v1beta1/showcase_v1beta1.yaml#L29

But this was to generate the Java related Mixin code.

File: showcase/pom.xml (hunk 3 above)
Comment (lqiu96) [reply showcase/pom.xml:83]: Yes. It was causing some errors in the ComplienceTest:
```
[ERROR] Errors:
[ERROR]   ComplianceClientHttpJsonTest.getIamPolicyTest » NotFound com.google.api.client.http.HttpResponseException: 404
GET https://localhost:7469/v1beta1/users/%5BUSER%5D/profile/blurbs/legacy/%5BLEGACY_USER%5D~%5BBLURB%5D:getIamPolicy
```

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java (hunk 4 above)
Comment (lqiu96) [reply showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:57]: I was using the IdentityClient to call the IAM RPCs which uses the User resource. I just gave it `user/{random_user_id}`.

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java (hunk 5 above)
Comment (lqiu96) [reply showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:96]: This is being thrown in the backend: https://github.com/googleapis/gapic-showcase/blob/b146f892d98ea0d733c45a2315941e8e6ab69c90/server/services/iam_policy_service.go#L27-L29

There is only a few validation checks that the backend iam server does.

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java (hunk 6 above)
Comment (lqiu96) [reply showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:156]: This was mapped from the showcase server's error responses: https://github.com/googleapis/gapic-showcase/blob/b146f892d98ea0d733c45a2315941e8e6ab69c90/server/services/iam_policy_service.go#L27-L29

File: showcase/pom.xml (hunk 3 above)
Comment (blakeli0) [reply showcase/pom.xml:83]: I see, I think https://github.com/googleapis/sdk-platform-java/issues/1547 at least worth investigating, then we can decide should we fix it or not. It could be that the showcase proto is configured in a way that real protos do not, if not, it could happen when a service team configures their protos in the same way as showcase protos.

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java (hunk 6 above)
Comment (blakeli0) [reply showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:156]: If the error is from the backend, I don't think we want to test them, or at most test one of them to make sure exceptions from IAM server can be properly mapped. Test all the permutations is not needed.

Comment (lqiu96) [reply showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:156]: Sounds good. I'll keep one test and remove the rest.

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java (hunk 4 above)
Comment (blakeli0) [reply showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java:56]: Discussed offline that this could be any format.

**Review by blakeli0 — APPROVED** [review]
